pkg/dogstatsd/listeners/listeners.go

```diff
@@ -0,0 +1,7 @@
+package listeners
+
+import "github.com/DataDog/datadog-agent/pkg/config"
+
+// GlobalPacketPool is used by the packet assembler to retrieve already allocated
+// buffer in order to avoid allocation.
+var GlobalPacketPool *PacketPool = NewPacketPool(config.Datadog.GetInt("dogstatsd_buffer_size"))
```

Comment:
Are we confident this code cannot be called before `dogstatsd_buffer_size` is set?

Reply:
➕ I don't think this will work as expected, `config.Datadog` is not fully initialized when this is called.

Reply:
Good point 🤔 I'll look into that